src/main/java/jenkins/plugins/git/GitSCMFileSystem.java

```diff
@@ -319,11 +324,17 @@ public SCMFileSystem build(@NonNull Item owner, @NonNull SCM scm, @CheckForNull
                 } catch (URISyntaxException ex) {
                     listener.getLogger().println("URI syntax exception for '" + remoteName + "' " + ex);
                 }
+                String prefix = Constants.R_HEADS; 
+                if(branchSpec.getName().startsWith(Constants.R_TAGS)){
+                    prefix = Constants.R_TAGS; 
+                }
                 String headName;
                 if (rev != null) {
                     headName = rev.getHead().getName();
                 } else {
-                    if (branchSpec.getName().startsWith(Constants.R_HEADS)) {
+                    if (branchSpec.getName().startsWith(Constants.R_TAGS)){
+                        headName = branchSpec.getName().substring(Constants.R_TAGS.length()); 
+                    } else if (branchSpec.getName().startsWith(Constants.R_HEADS)) {
```

Comment:
This method might be more generic using directly `prefix`, since it's containing `Constants.R_TAGS` or `Constants.R_HEADS` and in both cases you are doing something like `headName = branchSpec.getName().substring(prefix.length());`

Reply:
good catch! updated accordingly